cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -224,6 +235,23 @@ protected NewTopic createMetricsTopicFromReporterConfig(CruiseControlMetricsRepo
     return newTopic;
   }
 
+  protected void createCruiseControlMetricsProducer(Properties producerProps) throws KafkaException {
+    CruiseControlMetricsUtils.retry(() -> {
+      try {
+        _producer = new KafkaProducer<>(producerProps);
+        return false;
+      } catch (KafkaException e) {
+        if (e.getCause() instanceof ConfigException
+                && e.getCause().toString().contains("No resolvable bootstrap urls given in bootstrap.servers")) {
+          // dns resolution may not be complete yet, let's retry again later
+          LOG.error("Unable to create Cruise Control metrics producer. ", e.getCause());
+          return true;
+        }
+        throw (KafkaException) e.getCause();
```

Comment:
Why is the cause being re-thrown instead of the original exception `e` which is of type `KafkaException`?

Reply:
Sure I will re-throw the original exception.